worker/Cargo.toml

```diff
@@ -16,6 +16,7 @@ clap 					= { version = "2.33", features = [ "yaml" ] }
 lazy_static             = "1.4.0"
 parking_lot             = "0.11.1"
 thiserror               = "1.0"
+dashmap                 = "4.0.2"
```

Comment:
added new dependency to get a concurrent hash map. First I wrote my own concurrent hash map with `std::collection::hash_map` and `RwLock` - but I soon realized that there are quite a few challenges for implementing it properly, so I decided to get this from a library